pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -186,6 +206,13 @@ rec {
 
         } // lib.optionalAttrs (hardeningDisable != [] || hardeningEnable != []) {
           NIX_HARDENING_ENABLE = enabledHardeningOptions;
+        } // lib.optionalAttrs (outputs' != [ "out" ]) {
+          outputs = outputs';
+        } // lib.optionalAttrs doCheck' {
+          doCheck = true;
```

Comment:
I rather always set this one way or the other. Makes `set -u` in bash less annoying in principle.

Reply:
But `setup.sh` inspects nullness of these variables, not their values ( =/ ). And I didn't change that part.

Reply:
Mmm let me say again how I think things are slowly in case I have it wrong.

Today:
- `foo = true;` -> in drv, `export foo=1`
- `foo = false` -> in drv, `# nothing`
- `# no attr` -> not in drv, `# nothing`

How I want:
- `foo = true;` -> in drv, `export foo=1`
- `foo = false` -> in drv, `export foo=`
- `# no attr` -> not in drv, `# nothing`

`foo=` is certainly a thing in bash, and based on http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html I think should be a thing in env vars too. The difference I care about is what happens with `$doCheck` and `set -u`. I'd like to get rid of the `${doCheck:-}` stuff we have today,

Both `$doCheck` `${doCheck:-}` *without* `set -u` work with today and what I want. I assume you meant I wanted `export doCheck=0` or something.

Reply:
BTW I'd also consider doing
```rust
let doCheck0 = doCheck && stdenv.hostPlatform == stdenv.buildPlatform; in
let doCheck = doCheck0; in ...
```

so that shadowing prevents using the wrong version, and we can just do `inherit doInstall;` here.

Reply:
> `foo = false` -> in drv, `# nothing`

I assumed
`foo = false` -> in drv, `export foo=0`

can we postpone that change for a later PR? It will rebuild everything again I don't want to restart all the builds, I want to finish building and test booting.

Reply:
Sounds good.